### PR TITLE
CON-1036 Moved the CorDapp sample from the SDK

### PR DESCRIPTION
**Pull request by filipesoliveira**
Moved the cordapp sample from Conclave SDK.
The dockerfile that is used for the cordapp is not included because it seems that the user can run the cordapp without it as long as they install Java 8 as indicated in the docs.

### CODE REVIEW

File: cordapp/LICENCE.txt [hunk 1]
```diff
@@ -0,0 +1,13 @@
+  Copyright 2016, R3 Limited.
```
Comment (shamsasari) [inline cordapp/LICENCE.txt:1]: We didn't add a LICENSE file to the hello world sample

**Review by shamsasari — COMMENTED** [review]
Are you saying that we never needed to the cordapp container in the first place? If we had updated the our ci scripts to use the conclave-build container and then install JDK 8 inside then it would have work as well for us?!

File: cordapp/README.md [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+
+## Usage
+
+Unlike most CorDapps this one will *only run on Linux* due to the need for an enclave. But don't fear: you can use 
+virtualisation to run it on Windows and macOS as well. For Windows and macOS, this will unfortunately require you to
+do some manual work to set up a docker container (instructions below).
+
+### Linux
+
+1. Download and unpack JDK 8 and set `JAVA_HOME` environment variable to point it.
```
Comment (anien) [inline cordapp/README.md]: to point 'to' it

File: cordapp/README.md [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+2. Run `./gradlew workflows:test`
+
+### macOS
+
+1. Download and install Docker Desktop if it isn't installed already. Ensure that it is running.
+2. Download and unpack JDK 11 (LTS) and set `JAVA_HOME` environment variable to point it. If you're already using
```
Comment (anien) [inline cordapp/README.md]: missing 'to' (point to it). Multiple places have this missing.

**Comment by shamsasari** [comment]
I assume the existing instructions worked for all platforms. I’ll leave it to @anien to approve.

**Comment by anien** [comment]
I left just minor comments. 
Re instructions and testing, it would be easier for us to test this once it's all merged and we test linux by updating and running our QA tests and macOs and Windows by running it manually.

File: cordapp/LICENCE.txt (hunk 1 above)
Comment (filipesoliveira) [reply cordapp/LICENCE.txt:1]: The LICENSE.txt has been removed. There is a LICENSE.txt in the root folder (https://github.com/R3Conclave/conclave-samples/blob/master/LICENSE).

Comment (filipesoliveira) [reply cordapp/LICENCE.txt:1]: I will add the License to the cordapp because it is different from the one in the root directory. But we do not need to add the License to the hello-world sample because it is the same as the one in the root directory.

The following image contains the diff result and the hash for the LICENSE from the hello-world sample and the LICENSE in the root directory:
image

**Comment by filipesoliveira** [comment]
> Are you saying that we never needed to the cordapp container in the first place? If we had updated the our ci scripts to use the conclave-build container and then install JDK 8 inside then it would have work as well for us?!

It is not possible to install packages inside the container if you create an unprivileged user. The cordapp and sdk-build containers create the users host_user_1000 or host_user_1007. And that prevents you from installing packages.

I have used conclave-build which logs you in as root and that way I was able to install JDK 8. I ran the cordapp tests and it worked fine. However, there is this comment from the dockerfile:
`# Set up a fake user. Without this System.getProperty("user.home") returns '?' which confuses Corda.
COPY root/scripts/setup-users.sh /scripts/setup-users.sh
RUN echo '%sudo ALL=(ALL) NOPASSWD:ALL' >> /etc/sudoers && /scripts/setup-users.sh
`

Summing up, my answer is I am no longer sure because I do not know how to trigger the issue that led someone to add the comment above.

File: cordapp/LICENCE.txt (hunk 1 above)
Comment (filipesoliveira) [reply cordapp/LICENCE.txt:1]: The license for the cordapp has been added.

**Comment by anien** [comment]
This is fine from my side as we will do full testing once it is merged. I can't comment on the docker container, so leaving that to @shamsasari.

**Review by shamsasari — APPROVED** [review]
